geth/jail/extensions/console/console.go

```diff
@@ -0,0 +1,157 @@
+package console
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/robertkrimen/otto"
+	"github.com/status-im/status-go/geth/jail/extensions"
+	"github.com/status-im/status-go/geth/node"
+)
+
+var (
+	// Stdwarn defines the default writer for the console.warn
+	// delivery call.
+	Stdwarn io.Writer = os.Stdout
+
+	// Stdout defines the default writer for the console.error
+	// delivery call.
+	Stdout io.Writer = os.Stdout
+
+	// Stderr defines the default writer for the console.error
+	// delivery call.
+	Stderr io.Writer = os.Stderr
+
+	// Stddebug defines the default writer for the console.debug
+	// delivery call.
+	Stddebug io.Writer = os.Stdout
+
+	// EventConsoleLog defines the event type for the console.log call.
+	EventConsoleLog = "vm.console.log"
+
+	// EventConsoleWarn defines the event type for the console.debug call.
+	EventConsoleWarn = "vm.console.warn"
+
+	// EventConsoleDebug defines the event type for the console.debug call.
+	EventConsoleDebug = "vm.console.debug"
+
+	// EventConsoleError defines the event type for the console.error call.
+	EventConsoleError = "vm.console.error"
+
+	_ = extensions.Register(func(vm *otto.Otto) error {
+		return vm.Set("console", map[string]interface{}{
+			"log":   consoleLog,
+			"error": consoleError,
+			"debug": consoleDebug,
+			"warn":  consoleWarn,
+		})
+	})
+)
+
+// consoleLog provides the function caller for handling console.log
+// calls as replacement for the default console.log function within a
+// otto.Otto VM instance.
+func consoleLog(fn otto.FunctionCall) otto.Value {
+
+	// Record provided values into store for delviery.
+	node.SendSignal(node.SignalEnvelope{
+		Type:  EventConsoleLog,
+		Event: convertArgs(fn.ArgumentList),
+	})
+
+	// TODO(alex): Decide if we should use original console.log here.
+	// Next print out the giving values.
```

Comment:
I am not sure what is the point of using original console log here, we are writing to stdout and signaling upwards - that's good enough imo.